app/javascript/dashboard/components/ui/DatePicker/components/CalendarDateInput.vue

```diff
@@ -0,0 +1,72 @@
+<script setup>
+import { defineProps, defineEmits } from 'vue';
+import { parseDateFromDMY } from '../helpers/DatePickerHelper';
+
+import { isValid, startOfDay, endOfDay, isBefore, isAfter } from 'date-fns';
+
+const props = defineProps({
+  calendarType: {
+    type: String,
+    default: '',
+  },
+  dateValue: {
+    type: String,
+    default: '',
+  },
+  compareDate: {
+    type: String,
+    default: '',
+  },
+  isDisabled: Boolean,
+});
+
+const emit = defineEmits(['update', 'validate', 'error']);
+
+const updateDate = newValue => {
+  emit('update', newValue);
+};
+
+const validateDate = dateString => {
+  const parsedDate = parseDateFromDMY(dateString);
+
+  if (!isValid(parsedDate)) {
+    emit('error', 'Please select a valid time range');
```

Comment:
Should we clear the input if the date is not valid?

Reply:
@muhsin-k No we don't clear the input, we only show the toast